pkg/sql/conn_executor_exec.go

```diff
@@ -252,6 +252,23 @@ func (ex *connExecutor) execStmtInOpenState(
 		}()
 	}
 
+	if ex.server.cfg.TestingKnobs.RoundTripsCount != nil &&
+		ex.server.cfg.TestingKnobs.RoundTripsCount.Stmt == stmt.SQL {
+		tr := ex.server.cfg.AmbientCtx.Tracer
+		var sp opentracing.Span
+		ctx, sp = tracing.StartSnowballTrace(ctx, tr, stmt.SQL)
+		// TODO(radu): consider removing this if/when #46164 is addressed.
+		p.extendedEvalCtx.Context = ctx
+
+		defer func() {
+			sp.Finish()
+			trace := tracing.GetRecording(sp)
+			if ex.server.cfg.TestingKnobs.RoundTripsCount != nil {
+				ex.server.cfg.TestingKnobs.RoundTripsCount.Count = trace.CountRoundTrips()
```

Comment:
It seems weird to mutate the testing knobs -- is this always guaranteed to work? Would having a function as the knob that takes in a trace make more sense?